pkg/skaffold/build/cluster/kaniko.go

```diff
@@ -77,10 +77,15 @@ func (b *Builder) buildWithKaniko(ctx context.Context, out io.Writer, workspace
 	}
 
 	pod, err := pods.Create(ctx, podSpec, metav1.CreateOptions{})
+
 	if err != nil {
 		return "", fmt.Errorf("creating kaniko pod: %w", err)
 	}
+
 	defer func() {
+		ctx, cancel := context.WithTimeout(context.Background(), 3*time.Second)
```

Comment:
Hi, can you add a comment to explains that if build interrupted the original context is cancelled and pod deletion will not be called, so we need a new ctx?

Reply:
Hi, yea, good idea.  Pushed changes